r/tests/testthat/test-dataset.R

```diff
@@ -194,6 +194,38 @@ test_that("Hive partitioning", {
   )
 })
 
+test_that("open_dataset fails when partitioning is passed as schema", {
+  expect_error(
+    open_dataset(hive_dir, hive_partition(other = utf8(), group = uint8()))
+  )
+
+  # skipping schema
+  ds <- open_dataset(hive_dir, , hive_partition(other = utf8(), group = uint8()))
```

Comment:
I don't think these next test cases are necessary, you're just testing how R matches function arguments, and they would test without your change.

Reply:
ready, removed